DNN Platform/Library/Services/OutputCache/Providers/MemoryProvider.cs

```diff
@@ -175,6 +187,99 @@ public override bool StreamOutput(int tabId, string cacheKey, HttpContext contex
         	return true;
         }
 
+        private void OnSynchronizeResponseCallback(IAsyncResult asynchronousResult)
+        {
+            // State of request is asynchronous.
+            var request = (HttpWebRequest)asynchronousResult.AsyncState;
+            try
+            {
+                var response = (HttpWebResponse)(request.EndGetResponse(asynchronousResult));
+
+                if (response == null || response.StatusCode != HttpStatusCode.OK)
+                {
+                    //Log Error
+                    Exceptions.Exceptions.LogException(new Exception("Synchronization Error: Url - " + request.RequestUri.AbsoluteUri + "; Status - " + response.StatusCode + " " + response.StatusDescription));
+                }
+            }
+            catch (WebException e)
+            {
+                if (e.Status != WebExceptionStatus.RequestCanceled)
+                {
+                    Exceptions.Exceptions.LogException(new Exception("Synchronization Error in Request: " + request.RequestUri.AbsoluteUri, e));
+                }
+            }
+            finally
+            {
+                _registeredWaitHandle?.Unregister(asynchronousResult.AsyncWaitHandle);
+            }
+        }
+
+        private static void TimeoutCallback(object state, bool timedOut)
+        {
+            if (timedOut)
+            {
+                var request = (HttpWebRequest)state;
+                request?.Abort();
```

Comment:
You don't need the null coalescing operator here. If you fear that can will fail you should do a say cast.